File: core/src/galenscovell/hinterstar/processing/GestureHandler.scala
package galenscovell.hinterstar.processing

import com.badlogic.gdx.input.GestureDetector
import galenscovell.hinterstar.ui.screens.GameScreen


class GestureHandler(root: GameScreen) extends GestureDetector.GestureAdapter {
  private val gameScreen: GameScreen = root


  override def pan(x: Float, y: Float, deltaX: Float, deltaY: Float): Boolean = {
    gameScreen.actionPan(deltaX / 10, -deltaY / 10)
    true
  }
}

File: core/src/galenscovell/hinterstar/processing/GestureHandler.kt
package galenscovell.hinterstar.processing

import com.badlogic.gdx.input.GestureDetector
import galenscovell.hinterstar.ui.screens.GameScreen


class GestureHandler(private val root: GameScreen) : GestureDetector.GestureAdapter() {



    override fun pan(x: Float, y: Float, deltaX: Float, deltaY: Float): Boolean {
        root.actionPan(deltaX / 10, -deltaY / 10)
        return true
    }
}